### PR TITLE
Let the DBs return lazy ByteStrings + fix readAll

### PR DESCRIPTION
**Pull request by mrBliss**

### CODE REVIEW

**Comment by edsko** [comment]
Closes #384.

File: ouroboros-consensus/src/Ouroboros/Storage/FS/API.hs [hunk 1]
```diff
@@ -136,36 +138,34 @@ hGetExactly :: forall m h
 hGetExactly hasFS h n = go n []
   where
     go :: Int -> [BS.ByteString] -> m BL.ByteString
```
Comment (edsko) [inline ouroboros-consensus/src/Ouroboros/Storage/FS/API.hs]: Nitpick: should be `Word` I guess..

File: ouroboros-consensus/src/Ouroboros/Storage/FS/API.hs [hunk 2]
```diff
@@ -136,36 +138,34 @@ hGetExactly :: forall m h
...
--- | Like hGetExactly, but it does not throw an exception if eof is found.
-hGetLenient :: forall m h
-            . (HasCallStack, Monad m)
-            => HasFS m h
-            -> h
-            -> Int
-            -> m BL.ByteString
-hGetLenient hasFS h n = go n []
+    go remainingBytes acc
+      | remainingBytes <= 0 = return $ BL.fromChunks $ reverse acc
```
Comment (edsko) [inline ouroboros-consensus/src/Ouroboros/Storage/FS/API.hs]: .. which would then mean that this should be `==`, no` <=`. Feel free to ignore though :)

**Comment by mrBliss** [comment]
@avieth just a heads-up: I'm updating the byron-proxy after changing the ImmutableDB to return lazy instead of strict bytestrings.

**Comment by avieth** [comment]
@mrBliss Oh good this is a feature I think I've wanted in the past. Will review today.

**Comment by mrBliss** [comment]
@avieth thanks!

**Pull request merged**